### PR TITLE
Update pysam recipe and patch cython run dependency

### PR DESCRIPTION
**Pull request by mencian**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by mencian** [comment]
@BiocondaBot please add label

File: recipes/pysam/build.sh [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+export CFLAGS="${CFLAGS} -O3 -DHAVE_LIBDEFLATE"
+export CPPFLAGS="${CPPFLAGS} -I${PREFIX}/include -DHAVE_LIBDEFLATE"
+export LDFLAGS="${LDFLAGS} -L$PREFIX/lib"
 
 # Fix for failing build of 0.20.0 on Linux
 # See https://github.com/pysam-developers/pysam/issues/1146
 platform=$(uname)
-if [[ "$platform" == "Linux" ]]; then
-    export CFLAGS="-fPIC $CFLAGS"
+if [[ "${platform}" == "Linux" ]]; then
```
Comment (jmarshall) [inline recipes/pysam/build.sh:13]: What is the reason for adding `{ … }` everywhere?

IMHO all it does is add noise in the git history. Please don't.

File: recipes/pysam/meta.yaml [hunk 2]
```diff
@@ -1,26 +1,31 @@
+{% set name = "pysam" %}
```
Comment (jmarshall) [inline recipes/pysam/meta.yaml:1]: Please don't extract `name` and `sha256` like this. It's pointless.

File: recipes/pysam/meta.yaml [hunk 3]
```diff
@@ -1,26 +1,31 @@
...
+  patches:
+    - cython.patch
 
 build:
-  number: 1
-  skip: True # [py2k]
-  binary_relocation: False # [linux]
+  number: 2
+  skip: True  # [py < 36]
+  binary_relocation: False  # [linux]
```
Comment (jmarshall) [inline recipes/pysam/meta.yaml:18]: Please don't change lines without good reason. It just adds noise in the git history.

File: recipes/pysam/meta.yaml [hunk 4]
```diff
@@ -29,17 +34,19 @@ requirements:
...
-    - openssl  # [not osx]
+
+test:
+  imports:
+    - pysam
 
 about:
-  home: https://github.com/pysam-developers/pysam
-  license: MIT
+  home: https://pypi.org/project/pysam/
```
Comment (jmarshall) [inline recipes/pysam/meta.yaml:43]: What is the reason for updating `home` away from the place where the upstream pysam maintainers work?

**Comment by jmarshall** [comment]
What is the reason for changing the build from `$PYTHON setup.py …` to `pip …`?

**Comment by mencian** [comment]
I used grayskull to generate a template for pysam, which uses `pip install` by default, then added a few changes of my own. I have removed all the unecessary changes and cleaned up redundant run dependencies in my recent commit.

**Pull request closed without merging**